test/framework/src/main/java/org/elasticsearch/test/AbstractBuilderTestCase.java

```diff
@@ -196,12 +198,17 @@ public static void afterClass() throws Exception {
 
     @Before
     public void beforeTest() throws IOException {
+        // the two following ranomized parameters are drawn every run, even though we use them in
+        // creating ServiceHolder just once. The other times are necessary so that further random
+        // generation stays the same for all test that is run, otherwise reproducability breaks
+        Settings indexSettings = createTestIndexSettings();
```

Comment:
The proposed fix is ok, but I wonder if the following wouldn't be better:

```
    @Before
    public void beforeTest() throws Exception {
        if (serviceHolder == null) {
            assert serviceHolderWithNoType == null;
            long masterSeed = SeedUtils.parseSeed(RandomizedTest.getContext().getRunnerSeedAsString());
            RandomizedTest.getContext().runWithPrivateRandomness(masterSeed, (Callable<Void>) () -> {
                serviceHolder = new ServiceHolder(nodeSettings, indexSettings(), getPlugins(), AbstractBuilderTestCase.this, true);
                serviceHolderWithNoType = new ServiceHolder(nodeSettings, indexSettings(), getPlugins(), AbstractBuilderTestCase.this, false);
                return null;
            });
        }
        serviceHolder.clientInvocationHandler.delegate = this;
        serviceHolderWithNoType.clientInvocationHandler.delegate = this;
    }
```

The idea is to execute this conditional initialization step under the suite randomness rather than the method randomness. If we don't want to parse the suite seed back from a string we could also generate our own suite seed, the important bit is that we generate it in the beforeClass which is executed under the suite randomness (master seed). I find that this way what we are doing is clearer. The proposed fix is good but can easily be reverted by mistake by just moving lines around.

Reply:
Great suggestion, I added this.